x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Subject.java

```diff
@@ -41,17 +41,17 @@ public enum Type {
         SERVICE_ACCOUNT,
     }
 
-    private final Version version;
+    private final TransportVersion version;
     private final User user;
     private final Authentication.RealmRef realm;
     private final Type type;
     private final Map<String, Object> metadata;
 
     public Subject(User user, Authentication.RealmRef realm) {
-        this(user, realm, Version.CURRENT, Map.of());
+        this(user, realm, TransportVersion.CURRENT, Map.of());
     }
 
-    public Subject(User user, Authentication.RealmRef realm, Version version, Map<String, Object> metadata) {
+    public Subject(User user, Authentication.RealmRef realm, TransportVersion version, Map<String, Object> metadata) {
```

Comment:
What does this version number actually mean in the security code?

Reply:
judging by when this is called I would suspect that this has the same meaning as a transport version in StreamInput
https://github.com/pgomulka/elasticsearch/blob/1db5baa41b5c933ead5cba9a848b9aa95e749546/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java#L155

once this PR is ready for review I will follow up with security team

Reply:
as discussed ofline: Version in Subject and UserToken is used for serialisation into a Base64 strings. It does not use transport protocol (serialisation is into ThreadContext or XContent) but is using StreamInput/StreamOutput api